src/__tests__/integration/data/thethar/thethar-contract-wrc.js

```diff
@@ -0,0 +1,403 @@
+(() => {
+    // src/thetAR/actions/common.ts
+    var isAddress = (addr) => /[a-z0-9_-]{43}/i.test(addr);
+    var hashCheck = async (validHashs, contractTxId) => {
+      const tx = await SmartWeave.unsafeClient.transactions.get(contractTxId);
+      let SrcTxId;
+      tx.get("tags").forEach((tag) => {
+        let key = tag.get("name", { decode: true, string: true });
+        if (key === "Contract-Src") {
+          SrcTxId = tag.get("value", { decode: true, string: true });
+        }
+      });
+      if (!SrcTxId || !isAddress(SrcTxId)) {
+        throw new ContractError("Cannot find valid srcTxId in contract Tx content!");
+      }
+      const srcTx = await SmartWeave.unsafeClient.transactions.getData(SrcTxId, { decode: true, string: true });
+      if (srcTx.length < 1e4 && validHashs.includes(calcHash(srcTx))) {
+        return true;
+      }
+      return false;
+    };
+    var calcHash = (string) => {
+      var hash = 0, i, chr;
+      if (string.length === 0)
+        return hash;
+      for (i = 0; i < string.length; i++) {
+        chr = string.charCodeAt(i);
+        hash = (hash << 5) - hash + chr;
+        hash |= 0;
+      }
+      return hash;
+    };
+    var selectWeightedTokenHolder = async (balances) => {
+      let totalTokens = 0;
+      for (const address of Object.keys(balances)) {
+        totalTokens += balances[address];
+      }
+      let sum = 0;
+      const r = await getRandomIntNumber(totalTokens);
+      for (const address of Object.keys(balances)) {
+        sum += balances[address];
+        if (r <= sum && balances[address] > 0) {
+          return address;
+        }
+      }
+      return void 0;
```

Comment:
What is the difference between `return void 0;` and `return;`?

Reply:
I believe that is a compiled version (i.e. version that was generated by some like esbuild or tsc compiler?).

I need to remove these files (an the rust junk), as they are not used effectively (Kuba was analysing the original issue on these files, but since then we've managed to isolate the original issue into a separate MRE).